### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by bjoernricks**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
 pre-commit = ["autohooks.plugins.black"]
 ```
 
+## Proposed Workflow
+
+Using [pipenv](https://pipenv.readthedocs.io/) allows to install all
+dependencies and tools with a specific version into a virtual easily removable
```
Comment (wiegandm) [inline README.md]:
```suggestion
dependencies and tools with a specific version into a virtual, easily removable
```

File: README.md [hunk 2]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+python environment. Therefore it's best to maintain **autohooks** also via
```
Comment (wiegandm) [inline README.md]:
```suggestion
Python environment. Therefore it's best to maintain **autohooks** also via
```

File: README.md [hunk 3]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+pipenv. Because it is not required to build or run your software, it should be
+[installed as a development dependency](#install-autohooks-python-package).
+Installing and [activating](#activating-the-git-hooks) autohooks doesn't
+run any check or formatting by itself actually. Therefore it is required to
```
Comment (wiegandm) [inline README.md]:
```suggestion
actually run any check or formatting by itself. Therefore, it is required to
```

File: README.md [hunk 4]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+[choose and install a plugin](#configure-plugins-to-be-run).
+
+If all these tasks have been resolved the developers are able to install
```
Comment (wiegandm) [inline README.md]:
```suggestion
If all these tasks have been resolved, the developers are able to install
```

File: README.md [hunk 5]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+and activate autohooks with only one single command from you projects git
+repository:
+
+```sh
+pipenv install --dev
+```
+
+Because virtual environment are used for all dependencies including
+autohooks, the linting, formatting, etc. can on by done when running
```
Comment (wiegandm) [inline README.md]: > can on by done

Not sure. Do you mean `can then be done`?

File: README.md [hunk 6]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+
+```sh
+$ cd myproject
+$ pipenv install --dev
+$ pipenv shell
+(myproject)$ git commit
+```
+
+The advantage of this process is, if the user is not running `git commit` within
+the active virtual environment autohooks and its plugins are not executed.
```
Comment (wiegandm) [inline README.md]:
```suggestion
the active virtual environment, autohooks and its plugins are not executed.
```

File: README.md [hunk 7]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+
+```sh
+$ cd myproject
+$ git commit
+```
+
+This allows the user to choose whether to execute the hooks or to ignore them by
+activating the virtual environment.
```
Comment (wiegandm) [inline README.md]: Maybe rephrase this sentence? I know what is meant, but

> ignore them by activating the virtual environment.

can be easily misunderstood.

File: README.md (hunk 5 above)
Comment (bjoernricks) [reply README.md]: Should be `can only be done`

File: README.md (hunk 7 above)
Comment (bjoernricks) [reply README.md]: that's what this PR is for :grin: i want to avoid bad english or mis-phrased sentences.

File: README.md [hunk 8]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+dependencies and tools with a specific version into a virtual, easily removable
+Python environment. Therefore it's best to maintain **autohooks** also via
+pipenv. Because it is not required to build or run your software, it should be
+[installed as a development dependency](#install-autohooks-python-package).
+Installing and [activating](#activating-the-git-hooks) autohooks doesn't
+actually run any check or formatting by itself. Therefore, it is required to
+[choose and install a plugin](#configure-plugins-to-be-run).
+
+If all these tasks have been resolved, the developers are able to install
+and activate autohooks with only one single command from you projects git
```
Comment (swaterkamp) [inline README.md]:
```suggestion
and activate autohooks with only one single command from your project's git
```

File: README.md [hunk 9]
```diff
@@ -86,6 +96,47 @@ requires = ["setuptools", "wheel"]
...
+repository:
+
+```sh
+pipenv install --dev
+```
+
+Because virtual environment are used for all dependencies including
```
Comment (swaterkamp) [inline README.md]:
```suggestion
Because virtual environments are used for all dependencies including
```
